channels/test/liveserver.py

```diff
@@ -0,0 +1,205 @@
+import multiprocessing
+import os
+import sys
+
+import django
+from daphne.server import Server
+from django.core.exceptions import ImproperlyConfigured
+from django.test.testcases import TransactionTestCase
+from django.test.utils import modify_settings, override_settings
+from django.utils import six
+
+from .. import DEFAULT_CHANNEL_LAYER
+from ..asgi import ChannelLayerManager
+from ..worker import Worker, WorkerGroup
+
+# NOTE: We use ChannelLayerManager to prevent layer instance sharing
+# between forked process.  Some layers implementations create
+# connections inside the __init__ method.  After forking child
+# processes can lose the ability to use this connection and typically
+# stuck on some network operation.  To prevent this we use new
+# ChannelLayerManager each time we want to initiate default layer.
+# This gives us guaranty that new layer instance will be created and
+# new connection will be established.
+
+
+class WorkerProcess(multiprocessing.Process):
+
+    def __init__(self, is_ready, n_threads, overridden_settings,
+                 modified_settings):
+
+        self.is_ready = is_ready
+        self.n_threads = n_threads
+        self.overridden_settings = overridden_settings
+        self.modified_settings = modified_settings
+        super(WorkerProcess, self).__init__()
+        self.daemon = True
+
+    def run(self):
+
+        try:
+            django.setup(**{'set_prefix': False}
+                         if django.VERSION[1] > 9 else {})
+
+            if self.overridden_settings:
+                overridden = override_settings(**self.overridden_settings)
+                overridden.enable()
+            if self.modified_settings:
+                modified = modify_settings(self.modified_settings)
+                modified.enable()
+
+            channel_layers = ChannelLayerManager()
+            channel_layers[DEFAULT_CHANNEL_LAYER].router.check_default()
+            if self.n_threads == 1:
+                self.worker = Worker(
+                    channel_layer=channel_layers[DEFAULT_CHANNEL_LAYER],
+                    signal_handlers=False,
+                )
+            else:
+                self.worker = WorkerGroup(
+                    channel_layer=channel_layers[DEFAULT_CHANNEL_LAYER],
+                    signal_handlers=False,
+                    n_threads=self.n_threads,
+                )
+            self.worker.ready()
+            self.is_ready.set()
+            self.worker.run()
+        except Exception:
```

Comment:
It was inspired with Django own LiveServerThread run method. I think it's reasonable.